src/util/apiUtils.js

```diff
@@ -165,6 +165,8 @@ export const buildRequestArgs = externalRequestOpts =>
 		const externalRequestOptsQuery = JSON.parse(JSON.stringify(externalRequestOpts));
 		externalRequestOptsQuery.url = encodeURI(`/${endpoint}`);
 
+		delete externalRequestOptsQuery.headers['cf-ray'];
```

Comment:
this works, but most of the 'dirty headers' from the incoming request are cleaned out in `parseRequest` around line 227 - ideally this would happen there as well so that we have a 'clean' set of headers from input to output. happy to merge so you can test it out, though